utils/exceptions.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <QException>
+#include <QDebug>
+
+namespace jbrush
+{
+// Inherits from QException and allows the client to specify a unique string message that
+// is displayed in Qt's debugging console when raised
+class GeneralException : public QException
+{
+protected:
+    QString message;
+public:
+    GeneralException() { message = "Unidentified exception ocurred"; }
```

Comment:
An exception should always give information about what happened. I don't see the point in throwing an exception that gives no information.
```suggestion
```